contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -410,33 +416,61 @@ abstract contract FunctionsSubscriptions is IFunctionsSubscriptions, IERC677Rece
     emit SubscriptionConsumerAdded(subscriptionId, consumer);
   }
 
-  // @inheritdoc IFunctionsSubscriptions
-  function cancelSubscription(uint64 subscriptionId, address to) external override {
-    _whenNotPaused();
-    _onlySubscriptionOwner(subscriptionId);
-    _onlySenderThatAcceptedToS();
-
-    if (pendingRequestExists(subscriptionId)) {
-      revert CannotRemoveWithPendingRequests();
-    }
-
-    _cancelSubscriptionHelper(subscriptionId, to);
-  }
+  // Overriden in FunctionsRouter.sol
+  function _getSubscriptionDepositDetails() internal virtual returns (uint16, uint72);
 
-  function _cancelSubscriptionHelper(uint64 subscriptionId, address to) private {
+  function _cancelSubscriptionHelper(uint64 subscriptionId, address toAddress, bool skipSubscriptionDeposit) private {
     Subscription memory subscription = s_subscriptions[subscriptionId];
     uint96 balance = subscription.balance;
+    uint64 completedRequests = 0;
+
     // NOTE: loop iterations are bounded by config.maxConsumers
     // If no consumers, does nothing.
     for (uint256 i = 0; i < subscription.consumers.length; ++i) {
-      delete s_consumers[subscription.consumers[i]][subscriptionId];
+      address consumer = subscription.consumers[i];
+      completedRequests += s_consumers[consumer][subscriptionId].completedRequests;
+      delete s_consumers[consumer][subscriptionId];
     }
     delete s_subscriptions[subscriptionId];
-    s_totalLinkBalance -= balance;
 
-    IERC20(i_linkToken).safeTransfer(to, uint256(balance));
+    (uint16 subscriptionDepositCompletedRequests, uint72 subscriptionDepositJuels) = _getSubscriptionDepositDetails();
+
+    IERC20 linkToken = IERC20(i_linkToken);
```

Comment:
Not sure this improves code over casting a constant below. It's only ever used once to no need to assign to memory